atlassian_jwt_auth/contrib/django/decorators.py

```diff
@@ -30,43 +28,29 @@ def requires_asap_wrapper(request, *args, **kwargs):
             # that is the case to properly support Python 3.
             if isinstance(auth_header, six.string_types):
                 auth_header = auth_header.encode(encoding='iso-8859-1')
-            auth = auth_header.split(b' ')
-            if not auth or len(auth) != 2:
+            try:
+                scheme, auth = auth_header.split(b' ')
+            except ValueError:
+                scheme = b''
+
+            if scheme.lower() != b'bearer':
                 return HttpResponse('Unauthorized', status=401)
 
-            message = None
-            exception = None
             try:
-                asap_claims = parse_jwt(verifier, auth[1])
+                asap_claims = parse_jwt(verifier, auth)
                 verify_issuers(asap_claims, issuers)
                 request.asap_claims = asap_claims
-                return func(request, *args, **kwargs)
-            except RequestException as e:
-                # Error communicating to get key
-                message = 'Unauthorized: Communications error retrieving key'
-                exception = e
-            except PrivateKeyRetrieverException as e:
-                # Error parsing or getting private key
-                message = 'Unauthorized: Unable to retrieve private key'
-                exception = e
-            except PublicKeyRetrieverException as e:
-                # Error parsing or getting public key
-                message = 'Unauthorized: Unable to retrieve public key'
-                exception = e
-            except InvalidIssuerError as e:
-                message = 'Unauthorized: Invalid token issuer'
-                exception = e
+            except InvalidAudienceError as e:
+                # Audience mismatch
+                message = 'Forbidden: %s' % e
+                return HttpResponse(message, status=403)
```

Comment:
I prefer the previous response message because it provides details on why the request failed. Also, switching from `401` to `403` may seem useful but there seems to be a split about what to do in the case of an authorisation failure amongst asap libraries. For example, the go-asap library uses 401. Unless we have a good reason to change this I am inclined to say stick with `401`.